src/document/Document.js

```diff
@@ -758,6 +764,27 @@ define(function (require, exports, module) {
         return this.file instanceof InMemoryFile;
     };
 
+    /**
+     * Set lineEnding of the document. Change Text in the process.
+     */
+    Document.prototype.setLineEndings = function (lineEndings) {
+        if (this._lineEndings === lineEndings) {
+            return;
+        }
+        this._lineEndings = lineEndings;
```

Comment:
I tryed Notepad++ with a txt file. When you change the line ending, the first time goes in edited state, if you change back to the previous line ending it remains in an edited state.
So, to me, the prompt is expected.

The line ending changes should be part of the history but requires to add custom entry in CodeMirror. If someone at Adobe could do the courtesy it would be great.